api/src/neo4j/queries/search/globalSearch.js

```diff
@@ -28,11 +29,11 @@ UNWIND
       ELSE mids
   END AS mid
 MATCH (:Metabolite:${model} {id:mid})-[${version}]-(cm:CompartmentalizedMetabolite)
-WITH DISTINCT(cm.id) as cmid
+WITH DISTINCT(cm.id) as cmid, mid
```

Comment:
Is `cmid` for CompartmentalizedMetaboliteIDs and `mid` for `MetaboliteIDs`? Is that because we have previously ignored the potential MetaboliteIDs (IDs without the ending compartment letter) that caused the problem with zero score hit?

Reply:
yes, exactly. we need to keep the metabolites id:s from the first part of the search since they are the ones that get the scores from neo4j.